src/components/tooltip/daikin-tooltip.ts

```diff
@@ -194,83 +206,96 @@ export class DaikinTooltip extends LitElement {
     this._autoUpdateCleanup = null;
   }
 
-  private _changeOpenState(state: boolean) {
-    if (this.open === state) {
-      return;
-    }
-    if (
-      !this.dispatchEvent(
-        new CustomEvent("beforetoggle", {
-          detail: { open: this.open },
-          bubbles: true,
-          composed: true,
-          cancelable: true,
-        })
-      )
-    ) {
-      return;
+  private _handleClick() {
+    if (this.trigger === "click") {
+      this._isMouseOpened = !this._isMouseOpened;
     }
-    this.open = state;
   }
 
-  private _handleClick() {
-    if (this.closeOnClick) {
-      this._changeOpenState(false);
+  private _handleMouseEnter() {
+    if (this.trigger === "hover") {
+      this._isMouseOpened = true;
     }
   }
 
   private _handleMouseLeave() {
     if (this.trigger === "hover") {
-      this._changeOpenState(false);
+      this._isMouseOpened = false;
     }
   }
 
-  private _handleMouseEnter() {
-    if (this.trigger === "hover") {
-      this._changeOpenState(true);
+  private _handleFocusIn() {
+    this._isFocused = true;
+  }
+
+  private _handleFocusOut() {
+    this._isFocused = false;
+  }
+
+  private _handleToggle(event: ToggleEvent) {
+    if (reDispatch(this, event, new ToggleEvent("beforetoggle", event))) {
+      this.open = !this.open;
     }
```

Comment:
Always reference the source (SSoT).

```suggestion
    if (reDispatch(this, event, new ToggleEvent("beforetoggle", event))) {
      this.open = event.newState === "open";
    }
```